src/backend/renderer/gles/profiler.rs

```diff
@@ -0,0 +1,186 @@
+use super::ffi;
+
+const MAX_QUERIES: usize = 64 * 1024;
+
+pub struct GpuProfiler {
+    context: tracy_client::GpuContext,
+    pool: Vec<GpuTracepoint>,
+    index: usize,
+    tracepoints: Vec<ExitedGpuTracepoint>,
+}
+
+impl GpuProfiler {
+    pub fn new(gl: &ffi::Gles2) -> Self {
+        let mut queries = [0; MAX_QUERIES];
+        unsafe {
+            gl.GenQueriesEXT(queries.len() as ffi::types::GLsizei, queries.as_mut_ptr());
+        }
+        let pool = queries
+            .chunks_exact(2)
+            .map(|pairs| {
+                let (entered, exited) = unsafe { (*pairs.get_unchecked(0), *pairs.get_unchecked(1)) };
+                GpuTracepoint { entered, exited }
+            })
+            .collect::<Vec<_>>();
+
+        let client = tracy_client::Client::start();
+
+        let mut bits = 0;
+        unsafe { gl.GetQueryivEXT(ffi::TIMESTAMP_EXT, ffi::QUERY_COUNTER_BITS_EXT, &mut bits) };
+
+        let mut gpu_timestamp = 0;
+        unsafe { gl.GetInteger64v(ffi::TIMESTAMP_EXT, &mut gpu_timestamp) };
+
+        let context = client
+            .new_gpu_context(
+                Some("GlesRenderer"),
```

Comment:
When the compositor is used with multiple GPUs, this should probably have different names. Maybe the drm node, though vendor and driver can also be queried through EGL.